include/mega/types.h

```diff
@@ -680,58 +680,71 @@ class SyncConfig
     {
         TYPE_UP = 0x01, // sync up from local to remote
         TYPE_DOWN = 0x02, // sync down from remote to local
-        TYPE_DEFAULT = TYPE_UP | TYPE_DOWN, // Two-way sync
+        TYPE_TWOWAY = TYPE_UP | TYPE_DOWN, // Two-way sync
     };
 
-    SyncConfig() = default;
+    SyncConfig(std::string localPath,
+               const handle remoteNode,
+               const fsfp_t localFingerprint,
+               std::vector<std::string> regExps = {},
+               const Type syncType = TYPE_TWOWAY,
+               const bool syncDeletions = false,
+               const bool forceOverwrite = false);
 
-    SyncConfig(const Type syncType, const bool syncDeletions, const bool forceOverwrite)
-        : mSyncType{syncType}
-        , mSyncDeletions{syncDeletions}
-        , mForceOverwrite{forceOverwrite}
-    {}
+    // whether this sync is active
+    bool isActive() const;
+
+    // sets whether this sync is active
+    void setActive(bool active);
+
+    // returns the local path of the sync
+    const std::string& getLocalPath() const;
+
+    // returns the remote path of the sync
+    handle getRemoteNode() const;
+
+    // returns the local fingerprint
+    fsfp_t getLocalFingerprint() const;
+
+    // returns the regular expressions
+    const std::vector<std::string>& getRegExps() const;
 
     // whether this is an up-sync from local to remote
-    bool isUpSync() const
-    {
-        return mSyncType & TYPE_UP;
-    }
+    bool isUpSync() const;
 
     // whether this is a down-sync from remote to local
-    bool isDownSync() const
-    {
-        return mSyncType & TYPE_DOWN;
-    }
+    bool isDownSync() const;
 
     // whether deletions are synced
-    bool syncDeletions() const
-    {
-        switch (mSyncType)
-        {
-            case TYPE_UP: return mSyncDeletions;
-            case TYPE_DOWN: return mSyncDeletions;
-            case TYPE_DEFAULT: return true;
-        }
-        assert(false);
-        return true;
-    }
+    bool syncDeletions() const;
 
     // whether changes are overwritten irregardless of file properties
-    bool forceOverwrite() const
-    {
-        switch (mSyncType)
-        {
-            case TYPE_UP: return mForceOverwrite;
-            case TYPE_DOWN: return mForceOverwrite;
-            case TYPE_DEFAULT: return false;
-        }
-        assert(false);
-        return false;
-    }
+    bool forceOverwrite() const;
+
+    // serializes the object to a string
+    std::string serialize() const;
```

Comment:
You can try (in a separate branch from `develop`) to adjust `serialize` to be a const method, as well as receive a `const string &data` if you like. But I wouldn't change it if it takes more than 10 minutes and specially if it may introduce any kind of error ;)

Reply:
I'll put it on my todo list for later :)